libs/hooks/open-telemetry/src/lib/metrics/metrics-hook.spec.ts

```diff
@@ -0,0 +1,218 @@
+import { BeforeHookContext, EvaluationDetails, HookContext, StandardResolutionReasons } from '@openfeature/js-sdk';
+import {
+  AggregationTemporality,
+  DataPoint,
+  InMemoryMetricExporter,
+  MeterProvider,
+  MetricReader,
+  ScopeMetrics,
+} from '@opentelemetry/sdk-metrics';
+import { MetricsHook } from './metrics-hook';
+import opentelemetry from '@opentelemetry/api';
+import { FEATURE_FLAG } from '../constants';
+
+// no-op "in-memory" reader
+class InMemoryMetricReader extends MetricReader {
+  protected onShutdown(): Promise<void> {
+    return Promise.resolve();
+  }
+  protected onForceFlush(): Promise<void> {
+    return Promise.resolve();
+  }
+}
+
+describe(MetricsHook.name, () => {
+  let reader: MetricReader;
+
+  beforeAll(() => {
+    reader = new InMemoryMetricReader();
+    const provider = new MeterProvider();
+
+    provider.addMetricReader(reader);
+
+    // Set this MeterProvider to be global to the app being instrumented.
+    const successul = opentelemetry.metrics.setGlobalMeterProvider(provider);
+    expect(successul).toBeTruthy();
+  });
+
+  describe(MetricsHook.prototype.before, () => {
+    it('should ', async () => {
+      const FLAG_KEY = 'before-test-key';
+      const PROVIDER_NAME = 'before-provider-name';
+      const hook = new MetricsHook();
+      const mockHookContext: BeforeHookContext = {
+        flagKey: FLAG_KEY,
+        providerMetadata: {
+          name: PROVIDER_NAME,
+        },
+      } as BeforeHookContext;
+
+      hook.before(mockHookContext);
+      const result = await reader.collect();
+      expect(
+        hasDataPointMatching(
+          result.resourceMetrics.scopeMetrics,
+          `${FEATURE_FLAG}.evaluation_active_count`,
```

Comment:
minor -  I think it's good to refer from metric name definitions [1]  rather than building the name here 🤔 

[1] - https://github.com/open-feature/js-sdk-contrib/pull/448/files#diff-f30ef9873c378e800d96715a2341ab282109adbf72b5dc527ad24a74cfec8b97R20-R23

Reply:
Did this.